src/handlers/check_commits/no_mentions.rs

```diff
@@ -0,0 +1,56 @@
+//! Purpose: When opening a PR, or pushing new changes, check for github mentions
+//! in commits and notify the user of our no-mentions in commits policy.
+
+use std::fmt::Write;
+
+use crate::{config::NoMentionsConfig, github::GithubCommit};
+
+pub(super) fn mentions_in_commits(
+    _conf: &NoMentionsConfig,
+    commits: &[GithubCommit],
+) -> Option<String> {
+    let mut mentions_commits = Vec::new();
+
+    for commit in commits {
+        if !parser::get_mentions(&commit.commit.message).is_empty() {
+            mentions_commits.push(&*commit.sha);
+        }
+    }
+
+    if mentions_commits.is_empty() {
+        None
+    } else {
+        Some(mentions_in_commits_warn(mentions_commits))
+    }
+}
+
+fn mentions_in_commits_warn(commits: Vec<&str>) -> String {
+    let mut warning =
+        format!("There are mentions (`@user`) in the following commits, please remove them.\n");
+
+    for commit in commits {
+        let _ = writeln!(warning, "    - {commit}");
+    }
+
+    warning
+}
+
+#[test]
+fn test_warning_printing() {
```

Comment:
It shouldn't be that hard to test the actual functionality more end-to-end. Could you please test `mentions_in_commits` instead of just `mentions_in_commits`? Should be enough to construct fake `GitHubCommit`s, we can add helper functions for doing that into `src/tests`, it will surely be useful in other places.

I want to add tests to new functionality or modifications if possible, and here I think it's a low-hanging fruit.

Reply:
Sure, added an end-to-end test, not sure about the quality of it, but works for our use-case.

Reply:
Thanks, much better than nothing!